src/controllers/hid/hidcontroller.cpp

```diff
@@ -243,13 +248,28 @@ int HidController::close() {
 bool HidController::poll() {
     Trace hidRead("HidController poll");
 
-    int result = hid_read(m_pHidDevice, m_pPollData, sizeof(m_pPollData) / sizeof(m_pPollData[0]));
-    if (result == -1) {
-        return false;
-    } else if (result > 0) {
-        Trace process("HidController process packet");
-        QByteArray outData(reinterpret_cast<char*>(m_pPollData), result);
-        receive(outData, mixxx::Time::elapsed());
+    int result = 1;
+    while (result > 0) {
+        // Rotate between two buffers so the memcmp below does not require deep copying to another buffer.
+        unsigned char* pCurrentBuffer = m_pPollData[m_iPollingBufferIndex];
+        m_iPollingBufferIndex = (m_iPollingBufferIndex + 1) % m_iNumBuffers;
+        unsigned char* pPreviousBuffer = m_pPollData[m_iPollingBufferIndex];
+
+        result = hid_read(m_pHidDevice, pCurrentBuffer, m_iBufferSize);
+        if (result == -1) {
+            return false;
+        } else if (result > 0) {
+            Trace process("HidController process packet");
+            // Some controllers such as the Gemini GMX continuously send input packets even if it
+            // is identical to the previous packet. If this loop processed all those redundant
+            // packets, it would be a big performance problem.
+            if (memcmp(pCurrentBuffer, pPreviousBuffer, m_iBufferSize) == 0) {
```

Comment:
This should be switchable by the mapping.

Reply:
I will not add complexity to the mapping system to avoid this trivial cost.

Reply:
memcmp'ing 255 bytes 10 million times takes 4.82 milliseconds with `-O0` on my machine. I think the cost is neglible. 

See: https://github.com/mixxxdj/mixxx/pull/2970#issuecomment-695837340

Reply:
Are we sure that omitting update suits to all cases? I think this can be surprising.